Sources/NIO/ChannelPipeline.swift

```diff
@@ -885,53 +934,17 @@ extension ChannelPipeline {
     /// - returns: A future that will be completed when all of the supplied `ChannelHandler`s were added.
     public func addHandlers(_ handlers: [ChannelHandler],
                             position: ChannelPipeline.Position = .last) -> EventLoopFuture<Void> {
-        var handlers = handlers
-        let individualPosition: ChannelPipeline.Position
-        switch position {
-        case .first:
-            handlers.reverse()
-            individualPosition = .first
-        case .last:
-            individualPosition = .last
-        case .before(let handler):
-            individualPosition = .before(handler)
-        case .after(let handler):
-            handlers.reverse()
-            individualPosition = .after(handler)
-        }
-
-        func addHandlersMakingPromise(handlers: [ChannelHandler],
-                                      transformedPosition: ChannelPipeline.Position) -> EventLoopFuture<Void> {
-            let promise = self.eventLoop.makePromise(of: Void.self)
-
-            // Add all the handlers.
-            func addAllHandlersAndComplete() {
-                for handler in handlers {
-                    let addResult = self._add(handler, position: transformedPosition)
-                    switch addResult {
-                    case .success:
-                        break // Keep going.
-                    case .failure:
-                        // Report failure and return.
-                        promise.completeWith(addResult)
-                        return
-                    }
-                }
-                promise.succeed(())
-            }
+        let promise = self.eventLoop.makePromise(of: Void.self)
 
-            if self.eventLoop.inEventLoop {
-                addAllHandlersAndComplete()
-            } else {
-                self.eventLoop.execute {
-                    addAllHandlersAndComplete()
-                }
+        if self.inEventLoop {
```

Comment:
There's some inconsistency in the use of this computed property or in asking the event loop. It'd be nice to be consistent.